### ISSUE TITLE
hist_option & hist_n need the save attribute

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/CMEPS#243]
With the latest CMEPS, I'm getting a failure in these two ERI tests:

```
    FAIL ERI_C2_Ly9.f09_g17_ais8gris4.T1850Gag.cheyenne_gnu RUN time=35
    FAIL ERI.f10_f10_mg37.I1850Clm50SpG.cheyenne_gnu.cism-test_coupling RUN time=40
```

with:

```
20211021 120338.386 ERROR            PET28 ESMF_Clock.F90:887 ESMF_ClockGetAlarm() Failure  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 /glade/work/sacks/cism-wrapper_code/cism-workspace2/components/cmeps/cime_config/../mediator/med_phases_history_mod.F90:235 Failure  - Passing error in return code
20211021 120338.386 ERROR            PET28 MED:src/addon/NUOPC/src/NUOPC_ModelBase.F90:2398 Failure  - Passing error in return code
20211021 120338.386 ERROR            PET28 /glade/p/cesmdata/cseg/PROGS/build/57992/esmf-8.2.0b13/src/Infrastructure/Trace/src/ESMCI_Trace.C:1512 ESMCI:TraceEventPhaseExit() Wrong argument specified  - Trace regions not properly nested exiting from region: [MED] med_phases_history_write Expected exit from: MED:(med_phases_history_write)
20211021 120338.386 ERROR            PET28 /glade/p/cesmdata/cseg/PROGS/build/57992/esmf-8.2.0b13/src/Infrastructure/Trace/src/ESMCI_Trace.C:1470 ESMCI::TraceEventCompPhaseExit() Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ESMCI_FTable.C:773 ESMCI_FTableCallEntryPointVMHop Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ESMCI_FTable.C:1041 c_esmc_compwait Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ESMF_Comp.F90:1256 ESMF_CompExecute Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ESMF_GridComp.F90:1903 ESMF_GridCompRun Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ESM0001:src/addon/NUOPC/src/NUOPC_Driver.F90:3320 Wrong argument specified  - Failed calling phase med_phases_history_write Run for modelComp 1
20211021 120338.386 ERROR            PET28 ESM0001:src/addon/NUOPC/src/NUOPC_Driver.F90:3587 Wrong argument specified  - Passing error in return code
20211021 120338.386 ERROR            PET28 ESM0001:src/addon/NUOPC/src/NUOPC_Driver.F90:3240 Wrong argument specified  - Passing error in return code
20211021 120338.386 ERROR            PET28 /glade/p/cesmdata/cseg/PROGS/build/57992/esmf-8.2.0b13/src/Infrastructure/Trace/src/ESMCI_Trace.C:1512 ESMCI:TraceEventPhaseExit() Wrong argument specified  - Trace regions not properly nested exiting from region: [ESM0001] RunPhase1 Expected exit from: MED:(med_phases_history_write)
20211021 120338.386 ERROR            PET28 /glade/p/cesmdata/cseg/PROGS/build/57992/esmf-8.2.0b13/src/Infrastructure/Trace/src/ESMCI_Trace.C:1470 ESMCI::TraceEventCompPhaseExit() Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ESMCI_FTable.C:773 ESMCI_FTableCallEntryPointVMHop Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ESMCI_FTable.C:1041 c_esmc_compwait Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ESMF_Comp.F90:1256 ESMF_CompExecute Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ESMF_GridComp.F90:1903 ESMF_GridCompRun Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.386 ERROR            PET28 ensemble:src/addon/NUOPC/src/NUOPC_Driver.F90:3320 Wrong argument specified  - Failed calling phase RunPhase1 Run for modelComp 1
20211021 120338.386 ERROR            PET28 ensemble:src/addon/NUOPC/src/NUOPC_Driver.F90:3587 Wrong argument specified  - Passing error in return code
20211021 120338.386 ERROR            PET28 ensemble:src/addon/NUOPC/src/NUOPC_Driver.F90:3240 Wrong argument specified  - Passing error in return code
20211021 120338.386 ERROR            PET28 /glade/p/cesmdata/cseg/PROGS/build/57992/esmf-8.2.0b13/src/Infrastructure/Trace/src/ESMCI_Trace.C:1512 ESMCI:TraceEventPhaseExit() Wrong argument specified  - Trace regions not properly nested exiting from region: [ensemble] RunPhase1 Expected exit from: MED:(med_phases_history_write)
20211021 120338.389 ERROR            PET28 /glade/p/cesmdata/cseg/PROGS/build/57992/esmf-8.2.0b13/src/Infrastructure/Trace/src/ESMCI_Trace.C:1470 ESMCI::TraceEventCompPhaseExit() Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.389 ERROR            PET28 ESMCI_FTable.C:773 ESMCI_FTableCallEntryPointVMHop Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.389 ERROR            PET28 ESMCI_FTable.C:1041 c_esmc_compwait Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.389 ERROR            PET28 ESMF_Comp.F90:1256 ESMF_CompExecute Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.389 ERROR            PET28 ESMF_GridComp.F90:1903 ESMF_GridCompRun Wrong argument specified  - Internal subroutine call returned Error
20211021 120338.390 INFO             PET28 (lnd_comp_nuopc):(ModelFinalize)  called
20211021 120338.390 INFO             PET28 (lnd_comp_nuopc):(ModelFinalize)  done
20211021 120338.391 INFO             PET28 (rof_comp_nuopc):(ModelFinalize)  called
20211021 120338.391 INFO             PET28 (rof_comp_nuopc):(ModelFinalize)  done
20211021 120338.416 INFO             PET28 ESMF_GridCompDestroy called
20211021 120338.416 INFO             PET28 ESMF_GridCompDestroy finished
20211021 120338.416 INFO             PET28 esmApp FINISHED
20211021 120338.416 INFO             PET28 Finalizing ESMF
```

This diff solves the problem, though I'm not sure if this is a correct or complete fix: I haven't looked closely but just wanted to get something working quickly:

```diff
diff --git a/mediator/med_phases_history_mod.F90 b/mediator/med_phases_history_mod.F90
index 1371253..9ffc894 100644
--- a/mediator/med_phases_history_mod.F90
+++ b/mediator/med_phases_history_mod.F90
@@ -144,8 +144,8 @@ contains
     type(ESMF_Clock)        :: mclock
     type(ESMF_Alarm)        :: alarm
     character(CS)           :: alarmname
-    character(CL)           :: hist_option  ! freq_option setting (ndays, nsteps, etc)
-    integer                 :: hist_n       ! freq_n setting relative to freq_option
+    character(CL), save     :: hist_option  ! freq_option setting (ndays, nsteps, etc)
+    integer, save           :: hist_n       ! freq_n setting relative to freq_option
     character(CL)           :: cvalue       ! attribute string
     logical                 :: isPresent
     logical                 :: isSet
```

**Comment by mvertens** [comment]
@billsacks @jedwards - Thanks for finding a temporary fix - however, this does not make sense to me. This is the first time we have needed to introduce the save attribute. @jedwards - can we please look at this together?

**Comment by billsacks** [comment]
/glade/work/sacks/cism-wrapper_code/cism-workspace2

**Comment by mvertens** [comment]
@billsacks - actually in looking at this subroutine some more, I agree that the save option should be there for the history variables. They are used later on subsequent calls. I'm amazed this worked before.

**Comment by mvertens** [comment]
This has been fixed in PR #249
